### PR TITLE
Add test and fix that repeated reconfigs works for schema changes.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@geirst Test failure is unrelated

File: searchcore/src/tests/proton/proton_config_fetcher/proton_config_fetcher_test.cpp [hunk 1]
```diff
@@ -267,16 +267,22 @@ TEST_FF("require that documentdb config manager builds schema with imported attr
...
+        const auto &schema = getDocumentDBConfig(f1, f2)->getSchemaSP();
+        EXPECT_EQUAL(1u, schema->getNumImportedAttributeFields());
+        EXPECT_EQUAL("imported", schema->getImportedAttributeFields()[0].getName());
+        EXPECT_EQUAL(1u, schema->getNumAttributeFields());
+        EXPECT_EQUAL("regular", schema->getAttributeFields()[0].getName());
+
+        const auto &attrCfg = getDocumentDBConfig(f1, f2)->getAttributesConfig();
+        EXPECT_EQUAL(1u, attrCfg.attribute.size());
+        EXPECT_EQUAL("regular", attrCfg.attribute[0].name);
+        docType->summaryBuilder.classes[0].id = i+2;
```
Comment (geirst) [inline searchcore/src/tests/proton/proton_config_fetcher/proton_config_fetcher_test.cpp]: Consider adding a comment on why this is needed

**Review by geirst — APPROVED** [review]
👍

File: searchcore/src/tests/proton/proton_config_fetcher/proton_config_fetcher_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/proton_config_fetcher/proton_config_fetcher_test.cpp]: Done

**Pull request merged**